README.md

```diff
@@ -1 +1,17 @@
 # brainglobe-workflows
+
+## Main purpose of brainglobe-workflows
+
+The main purpose of this repository is to host Python scripts and Command Line Interfaces for typical workflows that are easy for others to re-use. 
+These workflows typically combine several BrainGlobe tools (possibly together with other tools) to achieve a goal, 
+such as whole brain cell detection and atlas registration.
+
+## Secondary purposes of brainglobe-workflow, for developers
+
+Secondarily, this repository further has a related purposes for BrainGlobe code development itself. 
+* It contains benchmarks for these workflows to ensure performance is stable as the code changes
+    * (default) run benchmarks on CI with small data
+    * (optionally) specify a configuration (as a json file) to run benchmarks on locally available, typically "real", data
+    * these configurations are not intended to be user-facing
+
+Note that to catch dependency updates breaking our code early, we additionally run and benchmark the workflows locally on a desktop machine with large example data.
```

Comment:
This would be my suggestion for this second paragraph, but feel free to take or leave any bits you find more/less relevant

> ## Secondary purposes of brainglobe-workflows, for developers
> We also use these workflows to support code development. We do this by regularly benchmarking the time they take to complete to ensure performance is stable as the code changes.
> * Developers can install these benchmarks locally via `pip install [dev]`. By executing `asv run`,  the benchmarks will run with default parameters on a small dataset that is downloaded from [GIN](https://gin.g-node.org/G-Node/info/wiki). See [the asv docs](https://asv.readthedocs.io/en/v0.6.1/using.html#running-benchmarks) for further details on how to run benchmarks.
> * Developers can also run these benchmarks on  data available locally, by specifying the relevant paths in an input configuration file (JSON file).
> * We additionally run and benchmark the workflows locally on a internal desktop machine with large example datasets. These benchmarks are run periodically and the results are made publicly available.
>
> ### Future steps
> - Benchmarking memory usage
> - Run _relative_ benchmarks on small datasets on GitHub runners. This is useful to detect if a PR introduces performance regressions. We plan to follow an approach similar to the one proposed [here](https://labs.quansight.org/blog/2021/08/github-actions-benchmarks).

For now I direct the reader to the asv docs, but in the future we would have some quick steps here on how to run the benchmarks.

Reply:
This is great, but I think it's better to document future steps in issues.
* Opened https://github.com/brainglobe/brainglobe-workflows/issues/59 for memory usage
* #56 already exists for relative benchmarks

I'll fix the linting and add the reworked `Secondary purposes` section now.